lrlex/src/lib/ctbuilder.rs

```diff
@@ -463,29 +465,34 @@ where
         if let Some(inspect_lexerkind_cb) = self.inspect_lexerkind_cb {
             inspect_lexerkind_cb(lexerkind)?
         }
+        let header = Header::new();
         let (mut lexerdef, lex_flags): (Box<dyn LexerDef<LexerTypesT>>, LexFlags) = match lexerkind
         {
             LexerKind::LRNonStreamingLexer => {
-                let lexerdef = LRNonStreamingLexerDef::<LexerTypesT>::new_with_options(
-                    &lex_src,
-                    self.force_lex_flags.clone(),
-                    self.default_lex_flags.clone(),
-                )
-                .map_err(|errs| {
-                    errs.iter()
-                        .map(|e| {
-                            if let Some((line, column)) = line_cache.byte_to_line_num_and_col_num(
-                                &lex_src,
-                                e.spans().first().unwrap().start(),
-                            ) {
-                                format!("{} at line {line} column {column}", e)
-                            } else {
-                                format!("{}", e)
-                            }
-                        })
-                        .collect::<Vec<_>>()
-                        .join("\n")
-                })?;
+                let lexerdef =
+                    LRNonStreamingLexerDef::<LexerTypesT>::new_with_header(&lex_src, header)
+                        .map_err(|errs| {
+                            errs.iter()
+                                .map(|e| match e {
+                                    LexBuildHeaderError::Build(e) => {
+                                        if let Some((line, column)) = line_cache
+                                            .byte_to_line_num_and_col_num(
+                                                &lex_src,
+                                                e.spans().first().unwrap().start(),
+                                            )
+                                        {
+                                            format!("{} at line {line} column {column}", e)
+                                        } else {
+                                            format!("{}", e)
+                                        }
+                                    }
+                                    LexBuildHeaderError::Header(e) => {
+                                        format!("{}", e)
```

Comment:
Dumb question: can we not give line number information for the header?

Reply:
I think even that would be awkward, e.g. `YaccGrammarError` assumes spans exist (the least `SpansKind` assumes there is one span), we could add a `SpansKind::None`, and `YaccGrammarErrorKind::Header(HeaderErrorKind)`, It feels to me somewhat like throwing the baby out with the bath water though, and e.g. we still have to make changes of similar scope to adding `Location`.  In that sense `Location::CommandLine` is somewhat type equivalent to not having a span.